### PR TITLE
Nick Rui submission

### PR DESCRIPTION
**Pull request by nick-rui**
Final validation loss was 3.10031.

https://api.wandb.ai/links/nickrui-stanford-university/9dhkgy43

What I did: 
- Used torch.compile on default mode
- Used bfloat16 instead of float32
- Scaled up to d_model = 1024, d_ff = 4096, num_layers = 16, num_heads = 8
- Used squared ReLU FFN instead of SwiFLU
- Tied embedding matrix and LM head weights
- Implemented QK norm
- Implemented Muon optimizer for all 2D parameters except embedding/LM head
- Learning rates: used 9e-2 for Muon and 9e-3 for AdamW
- LR schedule: same for both optimizers, 1500 linear warmup, rest cosine annealing (scheduled to 9000 total steps, but 45 min timeout occurs around ~7500) to 0.1x the max lr.
- Weight decay 0.01, betas (0.9, 0.95), clip gradients to max l2 norm of 1.0
- Batch size 256, context length 512

### CODE REVIEW

**Comment by hermabr** [comment]
We will merge this once you add more details:))

**Comment by nick-rui** [comment]
updated with details!
